codice/ddf
# DDF-6386 Add support for source id and metacard type for csv metacard transforms

#### What does this PR do?
Forward port and conflict resolution of #6387 :

Downstream projects need to create CSV exports that include the source identifier and metacard type name. The current code only supports exporting data that is contained in a metacard attribute. However, the source identifier and metacard type name are stored separately from the attributes.

This PR also creates a constant in MetacardType for the string "metacard-type", which was being defined multiple times in DDF and in downstream projects.

This PR also includes a forward port of #6501 which adds UTC dates when exporting metacards in CSV format.

#### Who is reviewing it? 
@glenhein 
@jlcsmith 
@jrnorth 

#### Select relevant component teams: 

@codice/core-apis 
@codice/data 


#### Ask 2 committers to review/merge the PR and tag them here.

@andrewkfiedler
@clockard

#### How should this be tested?
Build with a downstream project that provides a UI capable of creating a CSV export. The export should include the source id and metacard type. Make sure that the exported data includes the correct values for these fields.

Export a metacard as a CSV file. Verify that the generated CSV uses UTC dates. For example:
![image](https://user-images.githubusercontent.com/29106944/185506247-f48ce2ac-a63e-4816-913e-6cf1fba2d9a5.png)


#### Any background context you want to provide?

#### What are the relevant tickets?
Fixes: #6386 

#### Screenshots


#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
